UPGRADE-7.1.md

```diff
@@ -27,14 +59,65 @@ ExpressionLanguage
  * Deprecate passing `null` as the allowed variable names to `ExpressionLanguage::lint()` and `Parser::lint()`,
    pass the `IGNORE_UNKNOWN_VARIABLES` flag instead to ignore unknown variables during linting
 
+Form
+----
+
+ * Deprecate not configuring the `default_protocol` option of the `UrlType`, it will default to `null` in 8.0 (the current default is `'http'`)
+
 FrameworkBundle
 ---------------
 
  * Mark classes `ConfigBuilderCacheWarmer`, `Router`, `SerializerCacheWarmer`, `TranslationsCacheWarmer`, `Translator` and `ValidatorCacheWarmer` as `final`
+ * Deprecate the `router.cache_dir` config option, the Router will always use the `kernel.build_dir` parameter
+
+Intl
+----
+
+ * [BC BREAK] Extracted `EmojiTransliterator` to a separate `symfony/emoji` component, the new FQCN is `Symfony\Component\Emoji\EmojiTransliterator`.
+   You must install the `symfony/emoji` component if you're using the old `EmojiTransliterator` class in the Intl component.
+
+Mailer
+------
+
+ * Postmark's "406 - Inactive recipient" API error code now results in a `PostmarkDeliveryEvent` instead of throwing a `HttpTransportException`
 
 PropertyInfo
 ------------
 
+ * Deprecate the `Type` class, use `Symfony\Component\TypeInfo\Type` class of `symfony/type-info` component instead
+
+   *Before*
+   ```php
+   use Symfony\Component\PropertyInfo\Type;
+
+   // bool
+   $boolType = new Type(LegacyType::BUILTIN_TYPE_BOOL);
+   // bool|null
+   $nullableType = new Type(LegacyType::BUILTIN_TYPE_BOOL, nullable: true);
+   // array<int, string|null>
+   $arrayType = new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING, true));
+
+   $arrayType->getBuiltinType(); // returns "array"
+   $arrayType->getCollectionKeyTypes(); // returns an array with an "int" Type instance
+   $arrayType->getCollectionValueTypes()[0]->isNullable(); // returns true
+   ```
```

Comment:
This code is dense, we could add blank lines.
```php
use Symfony\Component\PropertyInfo\Type;

// bool
$boolType = new Type(LegacyType::BUILTIN_TYPE_BOOL);

// bool|null
$nullableType = new Type(LegacyType::BUILTIN_TYPE_BOOL, nullable: true);

// array<int, string|null>
$arrayType = new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING, true));

$arrayType->getBuiltinType(); // returns "array"
$arrayType->getCollectionKeyTypes(); // returns an array with an "int" Type instance
$arrayType->getCollectionValueTypes()[0]->isNullable(); // returns true
```

Reply:
This is on purpose. I'm not expecting many people to be using the PropertyInfo as a standalone component.

I didn't want to take up too much vertical space in the UPGRADE guide for something that is not important for 90% of the readers.

Reply:
Agree